### PR TITLE
Add NATION table definition for JDBC connectors

### PR DESCRIPTION
**Pull request by sanjay990**

### CODE REVIEW

File: tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/jdbc/tpch/JdbcTpchDataSource.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    private final TpchTable table;
+    private final List<JDBCType> columnTypes;
+    private final double scaleFactor;
+
+    public JdbcTpchDataSource(TpchTable table, List<JDBCType> columns, double scaleFactor)
+    {
+        checkArgument(scaleFactor > 0.0, "scaleFactor should be greater than zero");
+
+        this.table = requireNonNull(table, "table is null");
+        this.columnTypes = columns;
```
Comment (maciejgrzybek) [inline tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/jdbc/tpch/JdbcTpchDataSource.java]: `ImmutableList.copyOf(...)`

**Comment by amrutagokhale** [comment]
LGTM.

**Comment by rschlussel** [comment]
I'm not so familiar with tempto, but the hive table definitions are annotated with `@RepositoryTableDefinition` so that they end up in the TableDefinitionsRepository.  Do you know what that's used for?  Is that just a hive thing or should this be there to?

**Comment by cawallin** [comment]
I believe that's so the tables can be used in convention based tests. It probably would be good for it to be there, though Tempto might not deal with multiple tables named the same thing very well.

**Comment by sanjay990** [comment]
@rschlussel : Added `@RepositoryTableDefinition`

File: tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/jdbc/tpch/JdbcTpchTableDefinitions.java [hunk 2]
```diff
@@ -29,8 +30,9 @@
     public static final double DEFAULT_SCALE_FACTOR = 0.01;
     public static final ImmutableList<JDBCType> NATION_TYPES = ImmutableList.of(BIGINT, VARCHAR, BIGINT, VARCHAR);
 
+    @RepositoryTableDefinition
     public static final RelationalTableDefinition NATION_DDL =
```
Comment (rschlussel) [inline tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/jdbc/tpch/JdbcTpchTableDefinitions.java]: just call this NATION to match the naming scheme of the other table definitions.

**Pull request closed without merging**